--- chkrootkit-0.45/Makefile~	2005-02-22 00:13:55.000000000 +0100
+++ chkrootkit-0.45/Makefile	2005-02-23 16:01:36.942576240 +0100
@@ -35,36 +35,28 @@
 
 chklastlog:   chklastlog.c
 	${CC} ${CFLAGS} -o $@ chklastlog.c
-	@strip $@
 
 chkwtmp:   chkwtmp.c
 	${CC} ${CFLAGS} -o $@ chkwtmp.c
-	@strip $@
 
 ifpromisc:   ifpromisc.c
 	${CC} ${CFLAGS} ${LDFLAGS}  -D_FILE_OFFSET_BITS=64 -o $@ ifpromisc.c
-	@strip $@
 
 chkproc:   chkproc.c
 	${CC} ${LDFLAGS} -o $@ chkproc.c
-	@strip $@
 
 chkdirs:   chkdirs.c
 	${CC} ${LDFLAGS} -o $@ chkdirs.c
-	@strip $@
 
 check_wtmpx:   check_wtmpx.c
 	${CC} ${LDFLAGS} -o $@ check_wtmpx.c
-	@strip $@
 
 chkutmp:   chkutmp.c
 	${CC} ${LDFLAGS} -o $@ chkutmp.c
-	@strip $@
 
 
 strings-static:   strings.c
 	${CC} ${STATIC} ${LDFLAGS} -o $@ strings.c
-	@strip $@
 
 clean:
 	rm -f ${OBJS} core chklastlog chkwtmp ifpromisc chkproc chkdirs check_wtmpx strings-static chkutmp
