diff -urNp -x '*.orig' chkrootkit-0.52.org/chkrootkit chkrootkit-0.52/chkrootkit
--- chkrootkit-0.52.org/chkrootkit	2026-03-03 00:13:08.029230175 +0100
+++ chkrootkit-0.52/chkrootkit	2026-03-03 00:13:08.095991729 +0100
@@ -217,7 +217,7 @@ z2 () {
       return ${NOT_TESTED}
     fi
 
-    WTMP=`loc wtmp wtmp "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
+    WTMP=`loc wtmpx wtmpx "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
     LASTLOG=`loc lastlog lastlog "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
 
     if [ ! -f $WTMP -a ! -f $LASTLOG ]; then
@@ -259,7 +259,7 @@ wted () {
 	  fi
        fi
    else
-       WTMP=`loc wtmp wtmp "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
+       WTMP=`loc wtmpx wtmpx "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
 
        if [ "${EXPERT}" = "t" ]; then
           expertmode_output "/usr/bin/chkrootkit-chkwtmp -f ${WTMP}"
