diff -Nur chkrootkit-0.37.orig/chkrootkit chkrootkit-0.37/chkrootkit
--- chkrootkit-0.37.orig/chkrootkit	Tue Sep 17 01:03:11 2002
+++ chkrootkit-0.37/chkrootkit	Thu Sep 19 13:12:20 2002
@@ -125,22 +125,22 @@
        return ${NOT_TESTED}
     fi
 
-    if [ ! -x ./ifpromisc ]; then
-      echo "not tested: can't exec ./ifpromisc"
+    if [ ! -x /usr/bin/chkrootkit-ifpromisc ]; then
+      echo "not tested: can't exec ./chkrootkit-ifpromisc"
       return ${NOT_TESTED}
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "./ifpromisc"
+        expertmode_output "/usr/bin/chkrootkit-ifpromisc"
         return 5
     fi
     echo
-    ./ifpromisc
+    /usr/bin/chkrootkit-ifpromisc
 }
 
 z2 () {
-    if [ ! -x ./chklastlog ]; then
-      echo "not tested: can't exec ./chklastlog"
+    if [ ! -x /usr/bin/chkrootkit-chklastlog ]; then
+      echo "not tested: can't exec /usr/bin/chkrootkit-chklastlog"
       return ${NOT_TESTED}
     fi
 
@@ -148,31 +148,31 @@
     LASTLOG=`loc lastlog lastlog "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "./chklastlog -f ${WTMP} -l ${LASTLOG}"
+        expertmode_output "/usr/bin/chkrootkit-chklastlog -f ${WTMP} -l ${LASTLOG}"
         return 5
     fi
 
-    if ./chklastlog -f ${WTMP} -l ${LASTLOG}
+    if /usr/bin/chkrootkit-chklastlog -f ${WTMP} -l ${LASTLOG}
     then
       if [ "${QUIET}" != "t" ]; then echo "nothing deleted"; fi
     fi
 }
 
 wted () {
-    if [ ! -x ./chkwtmp ]; then
-      echo "not tested: can't exec ./chkwtmp"
+    if [ ! -x /usr/bin/chkrootkit-chkwtmp ]; then
+      echo "not tested: can't exec /usr/bin/chkrootkit-chkwtmp"
       return ${NOT_TESTED}
     fi
 
     if [ "$SYSTEM" = "SunOS" ]; then
-       if [ ! -x ./check_wtmpx ]; then
-          echo "not tested: can't exec ./check_wtmpx"
+       if [ ! -x /usr/bin/chkrootkit-check_wtmpx ]; then
+          echo "not tested: can't exec /usr/bin/chkrootkit-check_wtmpx"
        else
           if [ "${EXPERT}" = "t" ]; then
-             expertmode_output "./check_wtmpx"
+             expertmode_output "/usr/bin/chkrootkit-check_wtmpx"
               return 5
           fi
-          if ./check_wtmpx
+          if /usr/bin/chkrootkit-check_wtmpx
              then
              if [ "${QUIET}" != "t" ]; then \
                 echo "nothing deleted in /var/adm/wtmpx"; fi
@@ -183,11 +183,11 @@
     WTMP=`loc wtmp wtmp "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "./chkwtmp -f ${WTMP}"
+        expertmode_output "/usr/bin/chkrootkit-chkwtmp -f ${WTMP}"
         return 5
     fi
 
-    if ./chkwtmp -f ${WTMP}
+    if /usr/bin/chkrootkit-chkwtmp -f ${WTMP}
     then
       if [ "${QUIET}" != "t" ]; then echo "nothing deleted"; fi
     fi
@@ -225,15 +225,15 @@
 {
     if [ \( "${SYSTEM}" = "Linux"  -o \( "${SYSTEM}" = "FreeBSD" -a \
        ${V} -gt 43 \) \) -a "${ROOTDIR}" = "/" ]; then
-      if [ ! -x ./chkproc ]; then
-        echo "not tested: can't exec ./chkproc"
+      if [ ! -x /usr/bin/chkrootkit-chkproc ]; then
+        echo "not tested: can't exec /usr/bin/chkrootkit-chkproc"
         return ${NOT_TESTED}
       fi
 
       if [ "${EXPERT}" = "t" ]; then
          [ -r /proc/ksyms ] &&  ${egrep} -i adore < /proc/ksyms 2>/dev/null
          [ -d /proc/knark ] &&  ${ls} -la /proc/knark 2> /dev/null
-          expertmode_output "./chkproc -v"
+          expertmode_output "/usr/bin/chkrootkit-chkproc -v"
           return 5
       fi
 
@@ -248,7 +248,7 @@
          echo "Warning: Knark LKM installed"
       fi
 
-      if ./chkproc
+      if /usr/bin/chkrootkit-chkproc
       then
            if [ "${QUIET}" != "t" ]; then echo "nothing detected"; fi
       else
@@ -1293,18 +1293,18 @@
 
    if [ "${SYSTEM}" = "Linux" ]
    then
-      if [ ! -x ./strings ]; then
-        printn "can't exec ./strings-static, "
+      if [ ! -x /usr/bin/chkrootkit-strings ]; then
+        printn "can't exec /usr/bin/chkrootkit-strings, "
         return ${NOT_TESTED}
       fi
 
       if [ "${EXPERT}" = "t" ]; then
-          expertmode_output "./strings -a ${CMD}"
+          expertmode_output "/usr/bin/chkrootkit-strings -a ${CMD}"
           return 5
       fi
 
       ### strings must be a statically linked binary.
-      if ./strings-static -a ${CMD} > /dev/null 2>&1
+      if /usr/bin/chkrootkit-strings -a ${CMD} > /dev/null 2>&1
       then
          STATUS=${INFECTED}
       fi
