--- ./chkrootkit.org	Wed Feb 13 12:58:05 2002
+++ ./chkrootkit	Wed Feb 13 12:58:45 2002
@@ -105,7 +105,7 @@
       return ${NOT_TESTED}
     fi
 
-    WTMP=`loc wtmp wtmp "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
+    WTMP=`loc wtmpx wtmpx "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
     LASTLOG=`loc lastlog lastlog "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
 
     if [ "${EXPERT}" = "t" ]; then
@@ -141,7 +141,7 @@
        fi
     fi
 
-    WTMP=`loc wtmp wtmp "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
+    WTMP=`loc wtmpx wtmpx "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
 
     if [ "${EXPERT}" = "t" ]; then
         expertmode_output "./chkrootkit-chkwtmp -f ${WTMP}"
